src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/VariantsToTable.java

```diff
@@ -278,13 +333,15 @@ private void emitMoltenizedOutput(final List<String> record) {
      * @param vc                the VariantContext whose field values we can to capture
      * @return List of lists of field values
      */
-    private List<List<String>> extractFields(final VariantContext vc) {
+    protected List<List<String>> extractFields(final VariantContext vc) {
 
         final int numRecordsToProduce = splitMultiAllelic ? vc.getAlternateAlleles().size() : 1;
         final List<List<String>> records = new ArrayList<>(numRecordsToProduce);
 
         final int numFields;
-        final boolean addGenotypeFields = genotypeFieldsToTake != null && !genotypeFieldsToTake.isEmpty();
+        final boolean hasGFields = genotypeFieldsToTake != null && !genotypeFieldsToTake.isEmpty();
+        final boolean hasASGFields = asGenotypeFieldsToTake != null && !asGenotypeFieldsToTake.isEmpty();
+        final boolean addGenotypeFields = hasGFields || hasASGFields;
         if ( addGenotypeFields ) {
             numFields = fieldsToTake.size() + genotypeFieldsToTake.size() * samples.size();
```

Comment:
Technically speaking you want to add the number of allele specific fields to `numFields`, but it's not an error because numFields is just an initial list size. In my opinion getting the right initial list size here isn't worth four lines of code.